synapse/app/generic_worker.py

```diff
@@ -458,21 +458,14 @@ def start(config_options: List[str]) -> None:
         config.appservice.notify_appservices = False
 
     if config.worker.worker_app == "synapse.app.user_dir":
-        if config.server.update_user_directory:
+        if not config.worker.update_user_directory:
             sys.stderr.write(
-                "\nThe update_user_directory must be disabled in the main synapse process"
-                "\nbefore they can be run in a separate worker."
-                "\nPlease add ``update_user_directory: false`` to the main config"
-                "\n"
+                "\nThe user directory worker is not allowed to perform user directory updates per the config."
+                "\nPlease add or edit "
+                f'``update_user_directory_on: "{config.worker.worker_name}"`` in the config\n'
```

Comment:
errr this seems a bit of an odd constraint. why isn't the user directory worker allowed to do the user directory updates?

(the previous code was checking that user directory updates were disabled on the *main* process before they were enabled implicitly on a worker by setting `worker_app: synapse.app.user_dir`)

Reply:
also note: if you're updating the text of the warning, use a `"""` at the top rather than this clunky multiple-lines-of-`"` thing.

Reply:
Good point, should i turn this into a warning then, to log when `update_user_directory_on` isnt specified to the local worker (if it's a user_dir one)?